tests/check/signed_video_helpers.c

```diff
@@ -220,6 +220,26 @@ create_signed_nalus_int(const char *str, struct sv_setting settings, bool new_pr
   return list;
 }
 
+signed_video_t *
+generate_and_set_private_key_on_camera_side(SignedVideoCodec codec, sign_algo_t algo)
```

Comment:
I think you should move this to check_h26xsigned_auth.c and put it as a static function right before the tests. You can see examples in other places in the tests.
Then add more inputs to make it more test-specific.